### PR TITLE
Update memcached to 1.5.3

### PR DESCRIPTION
**Pull request by gongled**
I also made the following fixes for a `memcached` update:

- Added `listen` option for an IP binding support.
- Improved `systemd` unit file without `kaosv` dependency.

### CODE REVIEW

File: memcached/SOURCES/memcached.service [hunk 1]
```diff
@@ -4,9 +4,9 @@ Before=httpd.service
 After=network.target
 
 [Service]
-PIDFile=/var/run/memcached/memcached.pid
-ExecStart=/etc/init.d/memcached start
-ExecStop=/etc/init.d/memcached stop
+EnvironmentFile=-/etc/sysconfig/memcached
```
Comment (andyone) [inline memcached/SOURCES/memcached.service]: Looks like path to sysconfig file is invalid

File: memcached/memcached.spec [hunk 2]
```diff
@@ -58,9 +58,11 @@ Source3:                  %{name}.service
 
 BuildRoot:                %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 
-BuildRequires:            gcc automake
+BuildRequires:            gcc automake git which
```
Comment (andyone) [inline memcached/memcached.spec]: Why build now require `git` and `which`? Something changed in build process?

**Review by andyone — COMMENTED** [review]
Thanks for your PR.

File: memcached/memcached.spec (hunk 2 above)
Comment (gongled) [reply memcached/memcached.spec]: `git` is not required actually, but `which` is. Have a look at `autogen.sh` script:
```
# Try to locate a program by using which, and verify that the file is an
# executable
locate_binary() {
  for f in $@
  do
    file=`which $f 2>/dev/null | grep -v '^no '`
    [...]
  done
  [...]
}
```

Clean CentOS does not have `which` utility installed. So, I have added it to `BuildRequires` list.

File: memcached/SOURCES/memcached.service (hunk 1 above)
Comment (gongled) [reply memcached/SOURCES/memcached.service]: It works (`-` means "ignore errors"), but I removed minus in front of the path for consistency of the systemd units.

Comment (andyone) [reply memcached/SOURCES/memcached.service]: I didn't know that. Thanks for explanation.

File: memcached/memcached.spec (hunk 2 above)
Comment (andyone) [reply memcached/memcached.spec]: > Clean CentOS

Do you mean Minimal?

Comment (gongled) [reply memcached/memcached.spec]: Yes, minimal CentOS does not have `which` installed by default.

**Comment by andyone** [comment]
Ok, looks good for me. Merged.

**Pull request merged**